centreon/www/include/configuration/configObject/timeperiod/DB-Func.php

```diff
@@ -568,3 +386,245 @@ function createTimePeriodsExceptions(array $params): void
+ *
+ * @return bool
+ */
+function updateTimeperiodInAPI($tp_id = null): bool
+{
+    if (! $tp_id) {
+        return true;
+    }
+
+    global $form, $basePath;
+
+    $formData = $form->getSubmitValues();
+
+    try {
+        updateTimeperiodByApi($formData, $basePath);
+
+        return true;
+    } catch (Throwable $th) {
+        echo "<div class='msg' align='center'>" . _($th->getMessage()) . '</div>';
+
+        return false;
+    }
+}
+
+/**
+ * Make the API request to update a timeeperiod .
+ * @param array $formData
+ * @param string $basePath
+ *
+ * @throws LogicException
+ * @throws Exception
+ *
+ * @return void
+ */
+function updateTimeperiodByApi(array $formData, string $basePath): void
+{
+    $kernel = Kernel::createForWeb();
+    /** @var Router $router */
+    $router = $kernel->getContainer()->get(Router::class);
+    $client = new CurlHttpClient();
+
+    $payload = getPayloadForTimePeriod($formData);
+    $url = $router->generate(
+        'UpdateTimePeriod',
+        $basePath ? ['base_uri' => $basePath, 'id' => $formData['tp_id']] : [],
+        UrlGeneratorInterface::ABSOLUTE_URL,
+    );
+
+    $headers = [
+        'Content-Type' => 'application/json',
+        'Cookie' => 'PHPSESSID=' . $_COOKIE['PHPSESSID'],
+    ];
+    $response = $client->request(
+        'PUT',
+        $url,
+        [
+            'headers' => $headers,
+            'body' => json_encode($payload),
+        ],
+    );
+
+    if ($response->getStatusCode() !== 204) {
+        $content = json_decode($response->getContent(false));
+
+        throw new Exception($content->message ?? 'Unexpected return status');
+    }
+}
+
+/**
+ * @param int[] $timeperiods
+ *
+ * @return bool
+ */
+function deleteTimePeriodInAPI(array $timeperiods = []): bool
+{
+    global $basePath;
+
+   try {
+        deleteTimeperiodByApi($basePath, $timeperiods);
+
+        return true;
+    } catch (Throwable $th) {
+        echo "<div class='msg' align='center'>" . _($th->getMessage()) . '</div>';
+
+        return false;
+    }
+}
+
+/**
+ * @param string $basePath
+ * @param int[] $timePeriodIds
+ *
+ * @throws Exception
+ */
+function deleteTimePeriodByAPI(string $basePath, array $timePeriodIds): void
+{
+    $kernel = Kernel::createForWeb();
+    /** @var Router $router */
+    $router = $kernel->getContainer()->get(Router::class);
+    $client = new CurlHttpClient();
+
+    $headers = [
+        'Content-Type' => 'application/json',
+        'Cookie' => 'PHPSESSID=' . $_COOKIE['PHPSESSID'] . ';XDEBUG_SESSION=XDEBUG_KEY',
+    ];
+
+    foreach ($timePeriodIds as $id) {
+        $url = $router->generate(
+            'DeleteTimePeriod',
+            $basePath ? ['base_uri' => $basePath, 'id' => $id] : [],
+            UrlGeneratorInterface::ABSOLUTE_URL,
+        );
+
+        $response = $client->request('DELETE', $url, ['headers' => $headers]);
+
+        if ($response->getStatusCode() !== 204) {
+            $content = json_decode($response->getContent(false));
```

Comment:
json_decode can return null on failure, you can use [JSON_THROW_ON_ERROR] (https://www.php.net/manual/en/json.constants.php#constant.json-throw-on-error) flag to improve error handling ;)

Reply:
ok for me :)